app/_src/kubernetes-ingress-controller/guides/high-availability/fallback-config.md

```diff
@@ -0,0 +1,480 @@
+> **Note:** The Fallback Configuration is an opt-in feature. You must
+> enable it by setting `FallbackConfiguration=true` in the controller's feature
+> gates configuration.
+
+## How it works
+
+{{site.kic_product_name}} translates Kubernetes objects it gets from the Kubernetes API and pushes
+the translation result via Kong’s Admin API to {{site.base_gateway}} instances. However, issues
+can arise at various stages of this process:
+
+1. Admission Webhook: Validates individual Kubernetes objects against schemas and basic rules.
+2. Translation Process: Detects issues like cross-object validation errors.
+3. Kong Response: Kong rejects the configuration and returns an error associated with a specific object.
+
+Fallback Configuration is triggered when an issue is detected in the 3rd stage and provides the following benefits:
+- Allows unaffected objects to be updated even when there are configuration errors.
+- Automatically builds a fallback configuration that Kong will accept without requiring user intervention by
+  either:
+  - Excluding the broken objects along with its dependants.
+  - Backfilling the broken object along with its dependants using the last valid Kubernetes objects' in-memory cache (if `--use-last-valid-config-for-fallback` flag is set).
+- Enables users to inspect and identify what objects were excluded from or backfilled in the configuration using
+  diagnostic endpoints.
+
+Below table summarizes the behavior of the Fallback Configuration feature based on the configuration:
+
+| `FallbackConfiguration` feature gate value | `--use-last-valid-config-for-fallback` flag value | Behavior                                                                                                                                             |
+|--------------------------------------------|---------------------------------------------------|------------------------------------------------------------------------------------------------------------------------------------------------------|
+| `false`                                    | `false`/`true` (has no effect)                    | The last valid configuration is used as a whole to recover (if stored).                                                                              |
+| `true`                                     | `false`                                           | The Fallback Configuration is triggered - broken objects and their dependants are excluded.                                                          |
+| `true`                                     | `true`                                            | The Fallback Configuration is triggered - broken objects and their dependants are excluded and backfilled with their last valid version (if stored). |
+
+Below diagram illustrates how the Fallback Configuration feature works in detail:
+
+![fallback-config](/assets/images/products/kubernetes-ingress-controller/fallback-config.png "Fallback Configuration diagram")
```

Comment:
Should this be a mermaid diagram instead? cc @lena-larionova

Reply:
I gave Mermaid a go although I wouldn't say I like the results much 😅 Probably this could be styled to look more appealing:

```mermaid
flowchart TD
classDef sub opacity:0
classDef note fill:#ffd, stroke:#ccb
classDef externalCall fill:#9e8ebf
classDef startEnd fill:#545454

    A("`Update loop triggered`") --> B["`Generate Kubernetes objects' store snapshot to be passed to the Translator`"]
    B --> C["`Translator: generate Kong configuration based on the generated snapshot`"]
    C --> D["`Configure Kong Gateways using generated declarative configuration`"]
    D --> E{"`Configuration rejected?`"}
    E --> |No| G["`Store the Kubernetes objects' snapshot to be used as the last valid state`"]
    E --> |Yes| F["`Build a dependency graph of Kubernetes objects - using the snapshot`"]
    G --> H["`Store the declarative configuration to be used as the last valid configuration`"]
    H --> Z("`End of the loop`")
    F --> I["`Exclude an object along with all its dependants from the fallback Kubernetes objects snapshot`"]
    I --> J["`Add a previous valid version of the object along with its dependants' previous versions to the fallback snapshot`"]
    J --> K["`Translator: generate Kong configuration based on the fallback snapshot`"]
    K --> L["`Configure Kong Gateways using generated fallback declarative configuration`"]
    L --> M{"`Fallback configuration rejected?`"}
    M --> |Yes| N{"`Was the last valid configuration preserved?`"}
    N --> |Yes| O["`Configure Kong Gateways using the last valid declarative configuration`"]
    O --> Z
    N --> |No| Z
    M --> |No| P["`Store the fallback Kubernetes objects' snapshot to be used as the last valid state`"]
    P --> R["`Store the fallback declarative configuration to be used as the last valid configuration`"]

    subgraph subI [" "]
        I
        noteI[For every invalid object reported by the Gateway]
    end

    subgraph subJ [" "]
        J
        noteJ[Given there was a last valid Kubernetes objects' store snapshot preserved and the object is present]
    end

    class subI,subJ sub
    class noteI,noteJ note
    class D externalCall
    class A,Z startEnd
```

Reply:
Have you tried it in the docs site? We have custom styling

Reply:
It should, though it might be complicated to make all the shapes go in the right order. I can help turn this into mermaid.js format

Reply:
> Have you tried it in the docs site? We have custom styling
Yup, I've tried. Didn't look well to my eyes. 😢

Reply:
@lena-larionova I believe you can start from what I posted in the comment above, there's a copy button in the top-right corner of the rendered diagram).

Reply:
* upside: I got the thing styled and managed to have some of the nodes take up less space. Added it to the PR to test
* downside: The diagram is still massive, and I think we're close to the limit of how much smaller we can force it.

Honestly, I prefer the png condensed version of the diagram, but ideally we use mermaid for maintainability's sake.

Reply:
@lena-larionova Cannot argue - the mermaid variant still takes more space on display than png, but as you mentioned, it's better for maintainability's sake and I think it's more important. BTW the styling looks really cool, thanks for fixing that! 😄